insights/parsers/lsattr.py

```diff
@@ -0,0 +1,45 @@
+"""
+LsAttr - command ``lsattr <files>``
+===================================
+"""
+
+from insights.core import CommandParser
+from insights.core.plugins import parser
+from insights.specs import Specs
+from insights.core.dr import SkipComponent
+
+
+@parser(Specs.lsattr)
+class LsAttr(CommandParser, dict):
+    """
+    Parses the output of the ``lsattr <files>`` command.
+    It stores the mapping of the file and its attributes to a dict,
+    the key is the file path, the value is the attributes by removing
+    the useless dashes.
+
+    Sample output::
+
+        ---------------- ./grub2-tools-2.02-0.86.el7.x86_64.rpm
+        ------i--------- ./grub2-common-2.02-0.86.el7.noarch.rpm
+        ---------------- ./grub2-tools-minimal-2.02-0.86.el7.x86_64.rpm
+
+    Raises:
+        SkipComponent: when the dict is empty
+
+    Examples:
+        >>> './grub2-common-2.02-0.86.el7.noarch.rpm' in lsattr_obj
+        True
+        >>> lsattr_obj['./grub2-common-2.02-0.86.el7.noarch.rpm']
+        'i'
+    """
+    def parse_content(self, content):
+        for line in content:
+            items = line.split(None, 1)
+            if len(items) == 2:
+                attr = items[0].strip().replace('-', '')
+                # still store the file when the attribute is empty, in case
+                # a direcotry is checked and we can not know if the file is not
+                # in the dir, or the attribute is empty if we don't store it.
```

Comment:
1. Is a `directory` a good candidate filter for this `lsattr_files`? From the name, it seems that only `file` is acceptable. Will you want it allows `directory` to be `add_filter`ed?
2. It would be better to store the bad results with a specific `value` (e.g. "n/a"). E.g. `lsattr: No such file or directory while trying to stat a/e`. It is sometimes useful, in case the `path` doesn't exist on the host.

Reply:
@xiangce for 1, yes, a directory is acceptable, and if a directory is used, it will list all the files in the directory.  Or I rename it to make it more readable.
For the others, I'll update it.